client/collection_graph.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type CollectionGraph struct {
+	Revision int                          `json:"revision"`
+	Groups   map[string]map[string]string `json:"groups"`
+}
+
+func (c *Client) GetCollectionGraph() (CollectionGraph, error) {
+	url := fmt.Sprintf("%s/api/collection/graph", c.BaseURL)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+
+	collectionGraph := CollectionGraph{}
+	if err != nil {
+		return collectionGraph, err
+	}
+	if err := c.sendRequest(req, &collectionGraph); err != nil {
+		return collectionGraph, err
+	}
+
+	log.Printf("[DEBUG] Got collection graph '%+v'", collectionGraph)
+	return collectionGraph, nil
+}
+
+func (c *Client) UpdateCollectionGraph(cg CollectionGraph) (CollectionGraph, error) {
+	url := fmt.Sprintf("%s/api/collection/graph", c.BaseURL)
+
+	retries := 10
+	updated := CollectionGraph{}
+	var err_rr error
+	for retries >= 0 {
+		current_cg, err := c.GetCollectionGraph()
```

Comment:
```suggestion
		currentCG, err := c.GetCollectionGraph()
```
Happens when mixing languages ;)